src/lib/CollectStore.js

```diff
@@ -284,8 +284,16 @@ export default class CollectStore {
     // detect oauth case
     const isOAuth = !!account.oauth
 
+    function createDirectoryIfNecessary (folderPath) {
+      if (folderPath) {
+        return cozy.client.files.createDirectoryByPath(folderPath)
+      } else {
+        return Promise.resolve({_id: null})
```

Comment:
IMHO, no. For two reason:
 * The object passed as a parameter to `Promise.resolve()` comes from nowhere and has no meaning . If we follow this logic, when called the method `createDirectoryIfNecessary` says : "Yeah I created a folder and its id is `null`". It should say instead "I created no folder, it was no necessary, so I return you `null`.
 * It's better to explicitly set the `folder_id` to `null` when the promise returns `null`, instead of auto-magically use a mocked folder object, which do not rely to the database in anyway: 
```javascript
account.folder_id = returnedFolder ? returnedFolder._id : null
```